cartpole_en.rst

```diff
@@ -0,0 +1,218 @@
+Cartpole
+~~~~~~~~~~~~~~~~~~
+
+Overview
+=======
+The inverted pendulum problem is a classic control problem in reinforcement learning. Cartpole is a discrete control task in the inverted pendulum problem. In the game there is a car with a pole on it. The cart slides side to side on a smooth and frictionless track to keep the pole upright. As shown below.
+
+.. image:: ./images/cartpole.gif
+   :align: center
+
+Install
+====
+
+installation method
+--------
+
+The Cartpole environment is built into the gym, and you can install the gym directly. Its environment id is \``CartPole-v0``\.
+
+.. code:: shell
+
+    pip install gym
+    
+Verify installation
+--------
+
+Run the following command on the Python command line to verify that the installation is successful.
+
+.. code:: shell
+
+    import gym
+    env = gym.make('CartPole-v0')
+    obs = env.reset()
+    print(obs)
+
+Environment introduction
+=========
+
+Action space
+------------
+
+The action space of Cartpole belongs to the discrete action space, and there are two discrete actions, namely left shift and right shift.
+
+- \ ``Left Move`` \: 0 means to move the agent to the left.
+
+- \ ``Right move`` \: 1 means to move the agent to the right.
+
+Using the gym environment space definition can be expressed as:
+
+.. code::python
+    
+    action_space = spaces.Discrete(2)
+
+State space
+------------
+
+Cartpole's state space has 4 elements, which are:
+
+- \ ``Cart Position`` \: Cart position, in the range \ ``[-4.8, 4.8]`` \.
+  
+- \ ``Cart Velocity`` \: The speed of the cart, in the range \ ``[-inf, inf]`` \.
+
+- \``Pole Angle``\: The angle of the pole, in the range \``[-24 deg, 24 deg]``\.
+
+- \``Pole Angular Velocity``\: The angular velocity of the pole, in the range \``[-inf, inf]``\.
```

Comment:
- \ ``Pole Angular Velocity`` \: The angular velocity of the pole, in the range \ ``[-inf, inf]`` \.